### PR TITLE
Improve all-day event handling.  See issue #155, _make_all_day() issues.

### PR DESCRIPTION
**Pull request by perette**
This patch includes the set of changes for all-day events proposed in issue #155, except instead of adding a parameter to `make_all_day()` to revert to non-all-day events, I added a second member function, `cancel_all_day()`.  Corresponding unit tests are included.

### CODE REVIEW

File: ics/event.py [hunk 1]
```diff
@@ -217,27 +242,49 @@ def all_day(self):
         # the event may have an end, also given in 'day' precision
         return self._begin_precision == 'day'
 
+    def cancel_all_day(self):
```
Comment (C4ptainCrunch) [inline ics/event.py]: I'm not a fan of the naming, maybe doing `make_all_day(False)` like you suggested could be better

File: tests/event.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# vim: ts=4:sw=4:expandtab
```
Comment (C4ptainCrunch) [inline tests/event.py]: Could you remove this ? :)

File: tests/event.py [hunk 3]
```diff
@@ -81,6 +84,95 @@ def test_make_all_day2(self):
...
+
+        self.assertEqual(e._begin, begin)
+        self.assertEqual(e.begin, begin)
+
+        self.assertEqual(e._end_time, end)
+        self.assertEqual(e.end, end)
+
+        e.make_all_day()
+
+        self.assertEqual(e._begin, arrow.get ("2018-12-23"))
```
Comment (C4ptainCrunch) [inline tests/event.py]:
```suggestion
        self.assertEqual(e._begin, arrow.get("2018-12-23"))
```

File: tests/event.py [hunk 4]
```diff
@@ -81,6 +84,95 @@ def test_make_all_day2(self):
...
+        self.assertEqual(e.begin, arrow.get ("2018-12-23"))
```
Comment (C4ptainCrunch) [inline tests/event.py]:
```suggestion
        self.assertEqual(e.begin, arrow.get("2018-12-23"))
```

**Comment by perette** [comment]
I'm presently traveling, but plan to make the requested changes when I'm with a real machine instead of a toyblet.  ETA late June/early July/2-3 weeks.

**Comment by C4ptainCrunch** [comment]
I did not notice you pushed the changes i asked, sorry !
I fixed some conflicts and pushed to master in https://github.com/C4ptainCrunch/ics.py/commit/ed14cde096ad5b50bfa768c1fd32892eba744087. Your changes will be in 0.7 !

**Pull request closed without merging**